### PR TITLE
Example lab report

### PR DESCRIPTION
**Pull request by LuckyJosh**
First 4 days are done

the next 5 are WORK IN PROGRESS

To have a more or less realistic example report, I conducted a series of experiments (objects rolling down an inclined plane)
and wrote an almost[1] complete lab report for it. 

The overall idea is to have not only a somewhat realistic example but also building up this example.
Currently the final report is done and I'm going to split the final result into pieces,
that can be done as progressive exercises (essentially recreating the template piece by piece) with the final report as a comparison.

My rough idea of the steps are:

- [x] Day 1: Python 
   - Setup the folder an a python script
   - Some calculations with loops?
- [x] Day 2: Numpy, Matplotlib
   -  loading data with numpy
   - calculations with numpy
   - adding the plots (without fits)
- [x] Day 3: Scipy, uncertainties
   - adding the fits 
   - adding uncertainties
- [x] Day 4: unix, git
   -  turning the folder into a git repo (doing git stuff, push to github ...)
- [ ] Day 5: LaTeX-intro
   -  adding a first version of the document (maybe all in one file)
   -  splitting it up into multiple files
- [ ] Day 6: LaTeX-Math
   - Adding the equations to the respective sections
- [ ] Day 7: LaTeX-Floats and Bibliography
   - Adding the plots as figures 
   - Adding the tables by typing them out (not via the script)  
   - Adding the sources to the bib files
- [ ] Day 8: LaTeX + Matplotlib
   - adding header-matplotlib and matplotlibrc
- [ ] Day 9: Make
   - Adding the Makefile 

[1]  All sections do contain some 'real-world' examples for tables, images, equations
(single and multiline), and references to any of these. And the overall structure fits as well, but I did not find anything to write in the discussion section, that would make the example any better
(I just left a todo note).

### CODE REVIEW

**Comment by jpwgnr** [comment]
krass

File: exercises-latex/example-lab-report/loesung/v16516/content/durchfuehrung.tex [hunk 1]
```diff
@@ -0,0 +1,20 @@
+\section{Durchführung}
+\label{sec:Durchführung}
+
+Wie einst von Galileo Galilei \cite[232]{galilei1623} persönlich, wurden im hier beschriebenen Versuch
+Objekte (eine Kugel und  Holzylinder in Form eines Glases) eine schiefe Ebene hinab gerollt.
+
+Notiert wird dabei die Starthöhen $h$. Der Prozess des Herabrollens wird mit einer Kamera 
+aufgezeichnet, um aus die Zeit $t$ bis zum erreichen des Endes der schiefen Ebene möglichst genau
```
Comment (jpwgnr) [inline exercises-latex/example-lab-report/loesung/v16516/content/durchfuehrung.tex:8]: um ~aus~ die Zeit

File: exercises-latex/example-lab-report/loesung/v16516/latex_formatting.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from uncertainties.unumpy import (
+    std_devs as stds,
+)
+
+def make_qty(num, unit, exp="", figures=None):
+    """Format an uncertainties ufloat as a \qty quantity"""
+    if np.any(stds([num])):
+        if figures is None:
+            figures = ""
+        x = "{0:.{1:}uf}".format(num, figures).replace("/", "")
```
Comment (jpwgnr) [inline exercises-latex/example-lab-report/loesung/v16516/latex_formatting.py:11]: I would always choose for f-strings at this point and I think they are more "modern" and [here](https://realpython.com/python-f-strings/) they claim they are faster than `.format`

File: exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+Durch Mittelung der Messwerte für die selbe Höhe $h$ ergeben sich die in 
+\autoref{tab:averaged-measurements} berechnet und werden für die folgenden
+Auswertungsschritte verwendet.
+
+
+\begin{table}
+  \centering
+  \input{build/table_averaged-measurements.tex}
+  \caption{Für gleiche Starthöhe $h$ gemittelte Messwerte der Zeit $t$.}
```
Comment (jpwgnr) [inline exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex:46]: are table captions not usually above the content?

File: exercises-latex/example-lab-report/loesung/v16516/content/theorie.tex [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    \draw (0,0) -- (3,0) --node[below, near end]{$l$} (0,1) -- cycle;
+    \draw (0.25,1.19)node[below]{$m$} circle (0.25cm);
+    \draw[->] (0.25,1.19) -- (0.90, 0.94) node[above] {\small$\vec{v}$};
+    \draw[|-|] (-0.15,0.95) -- node[left]{$h$} (-0.15, 0);
+    \draw[dashed,opacity=0.65] (0.94,0.96) circle (0.25cm);
+    \draw[dashed,opacity=0.40] (1.75,0.68) circle (0.25cm);
+    \draw[dashed,opacity=0.25] (3.,0.26) circle (0.25cm);
+    \draw[->] (0.25, 1.03)  arc (270:50:0.16) node[below]{\small$\omega$};
+  \end{tikzpicture}
+  \caption{Schematische Darstellung des Versuchsaufbaus.}
```
Comment (jpwgnr) [inline exercises-latex/example-lab-report/loesung/v16516/content/theorie.tex:22]: I would add a couple more words on what is actually shown here

File: exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+\begin{align}
+  \label{eq:physical-properties}
+  m_\text{Z} &= \input{mass_ball.tex} & m_\text{K} &= \input{mass_ball.tex}\nonumber\\
+  r_{\text{Z},\text{i}} &= \input{../build/radius-inner_cylinder.tex} & r_\text{K} &= \input{radius_ball.tex}\\
+  r_{\text{Z},\text{a}} &= \input{../build/radius-outer_cylinder.tex} &&\nonumber
+\end{align}
+
+Berechnet wurde aus diesen Größen das jeweilige, theoretiche Trägheitsmoment
+
```
Comment (jpwgnr) [inline exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex:15]: I would not leave blank lines between equation and alignment environments, this creates weird spaces

File: exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+\begin{align}
+  \label{eq:moments-inertia}
+  I_\text{Z} &= \input{./../build/theoretical-I_cylinder.tex} && I_\text{K} = \input{./../build/theoretical-I_ball.tex}.
```
Comment (jpwgnr) [inline exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex:18]: I guess I would add a comma in here between the values

File: exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+\end{align}
+
+
+
+Aus den aufgenommenen Videos der Versuche wurden die in \autoref{tab:all-measurements} folgenden Frameindizes extrahiert.
+Die ebenfalls aufgeführte Zeitdauer $t$ wurde dann jeweils aus der aus dem Startframeindex $F_\text{i}$,
+dem Endframeindex $F_\text{f}$ und der Kameraframerate $\symup{fps}=\input{framerate.tex}$ wie folgt
```
Comment (jpwgnr) [inline exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex:25]: here I would have the framerate unit with a `per-mode=reciprocal`

File: exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+\begin{table}
+  \centering
+  \input{build/table_all-measurements.tex}
+  \caption{Alle aufgenommenen Werte, das heißt mit dreifach wiederholter Messungen je Höhe.}
+  \label{tab:all-measurements}
+\end{table}
+
+Durch Mittelung der Messwerte für die selbe Höhe $h$ ergeben sich die in 
+\autoref{tab:averaged-measurements} berechnet und werden für die folgenden
```
Comment (jpwgnr) [inline exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex:39]: "ergeben sich die in Tabelle 2 berechnet" here is something weird

File: exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex (hunk 6 above)
Comment (jpwgnr) [reply exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex:18]: also these values I would probably format in kg mm^2 as these have now many zeros which is not scientific

**Comment by jpwgnr** [comment]
couple of small things to improve, all in all it is really a beautiful example! Well done.

**Comment by LuckyJosh** [comment]
> couple of small things to improve, all in all it is really a beautiful example! Well done.

Thanks for the review :+1:

File: exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex (hunk 7 above)
Comment (maxnoe) [reply exercises-latex/example-lab-report/loesung/v16516/content/auswertung.tex]: symup is the wrong command for fps, it still treats it as f • p • s, just upright.

Correct would be operatorname

**Comment by LuckyJosh** [comment]
The first 4 days of 'steps' towards the protocol are now done.

**Comment by LuckyJosh** [comment]
Thinking about this, I don't like:
- how big this PR is right now (and going to be) and 
- that multiple files are duplicated

I'll post the todo list as an issue and going to change the structure of this PR in a new one.

**Pull request closed without merging**